src/main/java/de/muenchen/isi/domain/service/AbfrageStatusService.java

```diff
@@ -302,4 +299,89 @@ public UUID getAbfrageId(final MessageHeaders messageHeaders) throws EntityNotFo
         }
         return abfrageId;
     }
+
+    /**
+     * Entnimmt alle möglichen Statusänderungen aus dem aktuellen Status und den Authorities des Nutzers.
+     *
+     * @param id vom Typ {@link UUID} um die Abfrage zu finden
+     * @return Liste von {@link TransitionModel} welche möglich sind
+     * @throws EntityNotFoundException falls die Abfrage nicht gefunden wird
+     */
+    public List<TransitionModel> getStatusAbfrageEventsBasedOnStateAndRole(final UUID id)
+        throws EntityNotFoundException {
+        List<AuthoritiesEnum> userRolesAuthorities = AuthenticationUtils.getUserAuthorities();
+        List<StatusAbfrageEvents> possibleAbfrageEventsBasedOnRole = new ArrayList<>();
+        Map<AuthoritiesEnum, StatusAbfrageEvents> rolesAndMatchingEvent = getRolesAndEventsMap();
+        List<TransitionModel> possibleTransitionsModel = new ArrayList<>();
+        for (AuthoritiesEnum userRole : userRolesAuthorities) {
+            possibleAbfrageEventsBasedOnRole.add(rolesAndMatchingEvent.get(userRole));
+        }
+        List<StatusAbfrageEvents> matchingAbfrageEvents = new ArrayList<>(getStatusAbfrageEventsBasedOnState(id));
+        matchingAbfrageEvents.retainAll(possibleAbfrageEventsBasedOnRole);
+        for (StatusAbfrageEvents event : matchingAbfrageEvents) {
```

Comment:
Die For-SChleife und das anschließende sortieren besser mit einem Stream machen. Wo immer möglich Streams statt schleifen verwenden.

```java
return matchingAbfrageEvents.stream()
   .map(event -> {
            TransitionModel transitionModel = new TransitionModel();
            transitionModel.setUrl(event.getUrl());
            transitionModel.setButtonName(event.getButtonName());
            transitionModel.setIndex(event.getIndex());
})
.sort(Comparator.comparingInt(TransitionModel::getIndex))

```